milli/src/search/criteria/mod.rs

```diff
@@ -290,18 +292,29 @@ impl<'t> CriteriaBuilder<'t> {
                     Some(ref sort_criteria) => {
                         for asc_desc in sort_criteria {
                             criterion = match asc_desc {
-                                AscDescName::Asc(field) => Box::new(AscDesc::asc(
+                                AscDescName::Asc(Member::Field(field)) => Box::new(AscDesc::asc(
                                     &self.index,
                                     &self.rtxn,
                                     criterion,
                                     field.to_string(),
                                 )?),
-                                AscDescName::Desc(field) => Box::new(AscDesc::desc(
+                                AscDescName::Desc(Member::Field(field)) => Box::new(AscDesc::desc(
                                     &self.index,
                                     &self.rtxn,
                                     criterion,
                                     field.to_string(),
                                 )?),
+                                AscDescName::Asc(Member::Geo(point)) => Box::new(Geo::new(
+                                    &self.index,
+                                    &self.rtxn,
+                                    criterion,
+                                    point.clone(),
+                                )?),
+                                AscDescName::Desc(Member::Geo(_point)) => {
+                                    return Err(UserError::InvalidSortName {
+                                        name: "Sorting in descending order is currently not supported for the geosearch".to_string(),
```

Comment:
```suggestion
                                        name: "sorting in descending order is not supported for the geosearch".to_string(),
```

I am not sure we want to support it in the futur, that's not well defined.

Reply:
I updated the message